### ISSUE TITLE
Service `istiod` not found while deploying with kubeflow v1.1.0

### ISSUE THREAD

**Issue by ccll** [kubeflow/kubeflow#5445]
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]

Try to install kubeflow v1.1.0 in on-prem k8s cluster, failed with log:

```
$ kfctl apply -V -f kfctl_k8s_istio.v1.1.0.yaml
...
kfctl exited with error: failed to apply:  (kubeflow.error): Code 500 with message: kfApp Apply failed for kustomize:  (kubeflow.error): Code 500 with message: Apply.Run : error when creating "/tmp/kout494903682": Internal error occurred: failed calling webhook "validation.istio.io": Post "https://istiod.istio-system.svc:443/validate?timeout=30s": service "istiod" not found
```

And we can see that the 'istiod' service is never created.

```
$ kubectl get svc -n istio-system
NAME                     TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)                                                                                                                                      AGE
istio-citadel            ClusterIP   10.43.68.7      <none>        8060/TCP,15014/TCP                                                                                                                           23h
istio-galley             ClusterIP   10.43.237.49    <none>        443/TCP,15014/TCP,9901/TCP                                                                                                                   23h
istio-ingressgateway     NodePort    10.43.160.63    <none>        15020:32170/TCP,80:31380/TCP,443:31390/TCP,31400:31400/TCP,15029:30093/TCP,15030:31454/TCP,15031:30400/TCP,15032:31758/TCP,15443:32747/TCP   23h
istio-pilot              ClusterIP   10.43.103.208   <none>        15010/TCP,15011/TCP,8080/TCP,15014/TCP                                                                                                       23h
istio-policy             ClusterIP   10.43.68.242    <none>        9091/TCP,15004/TCP,15014/TCP                                                                                                                 23h
istio-sidecar-injector   ClusterIP   10.43.238.82    <none>        443/TCP,15014/TCP                                                                                                                            23h
istio-telemetry          ClusterIP   10.43.85.115    <none>        9091/TCP,15004/TCP,15014/TCP,42422/TCP                                                                                                       23h
prometheus               ClusterIP   10.43.200.89    <none>        9090/TCP
```

**What did you expect to happen:**

The deployment should success without error.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Kubeflow version: (version number can be found at the bottom left corner of the Kubeflow dashboard): `kfctl_k8s_istio.v1.1.0.yaml`
- kfctl version: (use `kfctl version`): `kfctl v1.1.0-0-g9a3621e`
- Kubernetes platform: (e.g. `minikube`) `rancher k3s on-prem cluster, single master, no workers`
- Kubernetes version: (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.3", GitCommit:"1e11e4a2108024935ecfcb2912226cedeafd99df", GitTreeState:"clean", BuildDate:"2020-10-14T12:50:19Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.4+k3s1", GitCommit:"2532c10faad43e2b6e728fdcc01662dc13d37764", GitTreeState:"clean", BuildDate:"2020-11-18T22:11:18Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}
```
- OS (e.g. from `/etc/os-release`):

Client host (running kfctl):
```
NAME="Pop!_OS"
VERSION="20.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 20.04 LTS"
VERSION_ID="20.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
LOGO=distributor-logo-pop-os
```

k3s server host:
```
NAME="Ubuntu"
VERSION="20.04.1 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.1 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

**Comment by ccll** [comment]
There is a similar issue #5075, the author reported while installing v1.1.0 instead of v1.0.2 the issue is gone, but apparently not the same case here.

**Comment by summerisc** [comment]
I am using v1.2.0 and saw the same error, any updates on the solution for this?

**Comment by DavidSpek** [comment]
@summerisc Are you also using k3s? Kubeflow 1.3 is about to be released very soon and will use a different installation method (not `kfctl`) so I believe it might be best to see if you also experience the problem with the upcoming release.

**Comment by huboqiang** [comment]
Hi @summerisc @ccll  , I met the same problem, and I solved it by removing existed istio v1.7 by istioctl x uninstall --purge .

You can try to unisntall istio v1.7 and re-apply the yaml file.
